src/firmware/app/world/firmware_robot.c

```diff
@@ -17,41 +34,329 @@ struct FirmwareRobot
+
+    const float battery_voltage = app_firmware_robot_getBatteryVoltage(robot);
+    const float curr_vx         = app_firmware_robot_getVelocityX(robot);
+    const float curr_vy         = app_firmware_robot_getVelocityY(robot);
+    const float orientation     = app_firmware_robot_getOrientation(robot);
+
+    const float dest_x           = pos_trajectory.x_position[trajectory_index];
+    const float dest_y           = pos_trajectory.y_position[trajectory_index];
+    const float dest_orientation = pos_trajectory.orientation[trajectory_index];
+    float dest[3]                = {dest_x, dest_y, dest_orientation};
+
+    const float curr_x = app_firmware_robot_getPositionX(robot);
+    const float curr_y = app_firmware_robot_getPositionY(robot);
+
+    const float dx = dest_x - curr_x;
+    const float dy = dest_y - curr_y;
+
+    float total_disp = sqrtf(dx * dx + dy * dy);
+    // Add a small number to avoid division by zero
+    float major_vec[2] = {dx / (total_disp + 1e-6f), dy / (total_disp + 1e-6f)};
+    float minor_vec[2] = {major_vec[0], major_vec[1]};
+    rotate(minor_vec, P_PI / 2);
+
+    PhysBot pb = app_physbot_create(curr_vx, curr_vy, curr_x, curr_y, orientation, dest,
+                                    major_vec, minor_vec);
+
+    const float dest_speed = pos_trajectory.linear_speed[trajectory_index];
+
+    // plan major axis movement
+    const double ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED = 3.0;
+    float max_major_a     = (float)ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED;
+    float max_major_v     = max_speed_m_per_s;
+    float major_params[3] = {dest_speed, max_major_a, max_major_v};
+    app_physbot_planMove(&pb.maj, major_params);
+
+    // plan minor axis movement
+    float max_minor_a = (float)ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED / 2.0f;
+    float max_minor_v = max_speed_m_per_s / 2.0f;
+    float minor_params[3] = {0, max_minor_a, max_minor_v};
+    app_physbot_planMove(&pb.min, minor_params);
+
+    // plan rotation movement
+    plan_move_rotation(&pb, app_firmware_robot_getVelocityAngular(robot));
+
+    float accel[3] = {0, 0, pb.rot.accel};
+
+    // rotate the accel and apply it
+    app_physbot_computeAccelInLocalCoordinates(
+        accel, pb, app_firmware_robot_getOrientation(robot), major_vec, minor_vec);
+
+    app_control_applyAccel(robot_constants, controller_state, battery_voltage,
+                           front_left_wheel, front_right_wheel, back_left_wheel,
+                           back_right_wheel, accel[0], accel[1], accel[2]);
+}
+
+void velocity_wheels_followPosTrajectory(const FirmwareRobot_t* robot,
+                                         PositionTrajectory_t pos_trajectory,
+                                         size_t trajectory_index, float max_speed_m_per_s)
+{
+    // Convert position trajectory to velocity trajectory
+    // TODO: Implement this function
```

Comment:
TODO? Are you planning to do this or link it to an issue?

Reply:
I'll make an issue and link it.